src/compiler/emitter.ts

```diff
@@ -2867,8 +2871,18 @@ namespace ts {
             if (name.autoGenerateKind === GeneratedIdentifierKind.Node) {
                 // Node names generate unique names based on their original node
                 // and are cached based on that node's id.
+                let flags: TempFlags;

```

Comment:
Rather than read twice, the approach we've taken in other places (such as with context flags in the parser) is to do something like this:
```ts
if (name.skipNameGenerationScope) {
  const savedTempFlags = tempFlags;
  popNameGenerationScope(/*node*/ undefined);
  const result = generateNameCached(getNodeForGeneratedName(name));
  pushNameGenerationScope(/*node*/ undefined);
  tempFlags = savedTempFlags;
}
else {
  return generateNameCached(getNodeForGeneratedName(name));
}
```